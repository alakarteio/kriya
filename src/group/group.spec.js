/* eslint-disable import/no-extraneous-dependencies, react/jsx-filename-extension */
/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import Group from './group'

const snap = ({ onClick = () => {}, ...rest }) => {
  const component = renderer.create(
    <Group {...rest} onClick={onClick} />,
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
}

describe('common/Group', () => {
  it('should have a default behaviour', () => snap({}))
  it('should add custom className', () => snap({ className: 'custom' }))
  it('should add custom style', () => snap({ style: { backgroundColor: 'red' } }))
  it('should add children', () => snap({ children: 'a child' }))

  it('should add vertical class when vertical group', () => snap({ vertical: true }))
})

/* eslint-enable import/no-extraneous-dependencies, react/jsx-filename-extension */
