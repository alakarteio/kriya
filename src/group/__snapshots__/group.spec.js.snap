// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Group should add children 1`] = `
<div
  className="group horizontal"
  style={Object {}}
>
  a child
</div>
`;

exports[`common/Group should add custom className 1`] = `
<div
  className="group custom horizontal"
  style={Object {}}
/>
`;

exports[`common/Group should add custom style 1`] = `
<div
  className="group horizontal"
  style={
    Object {
      "backgroundColor": "red",
    }
  }
/>
`;

exports[`common/Group should add vertical class when vertical group 1`] = `
<div
  className="group vertical"
  style={Object {}}
/>
`;

exports[`common/Group should have a default behaviour 1`] = `
<div
  className="group horizontal"
  style={Object {}}
/>
`;
